CodeU_Assignement_1/src/codeu_assignement_1/SinglyLinkedList.java

```diff
@@ -0,0 +1,236 @@
+/*
+ * To change this license header, choose License Headers in Project Properties.
+ * To change this template file, choose Tools | Templates
+ * and open the template in the editor.
+ */
+package codeu_assignement_1;
+
+import java.util.Collection;
+import java.util.Iterator;
+
+/**
+ *
+ * @author Karine
+ * 
+ * Bounds, index and general interface is as in Java LinkedList JavaDoc
+ * https://docs.oracle.com/javase/7/docs/api/java/util/LinkedList.html
+ * 
+ */
+public class SinglyLinkedList <E> {
+    /*  Inner pojo class Node in the list  */
+    public class Node <T>
+    {
+        protected T data;
+        protected Node next;
+
+        /*  Default C'tor  */
+        public Node()
+        {
+            next = null;
+            data = null;
+        }    
+        /*  C'tor  */
+        public Node(T d,Node n)
+        {
+            data = d;
+            next = n;
+        }    
+        // Setters
+        public void setNext(Node n) { next = n; }    
+        public void setData(T d) { data = d; } 
+        
+        // Getter
+        public Node getNext() { return next; }    
+        public T getData() { return data; }
+    }
+    
+    
+    /* Data members of LinkedList */
+    protected int size=0;
+    protected Node m_first=null;
+    protected Node m_last=null;
+    
+    // Default c'tor
+    public SinglyLinkedList(){} 
+
+    // C'tor: adds all items of another collection
+    public SinglyLinkedList(Collection<? extends E> c) { this.addAll(c); } 
+            
+    /*  Check if the list is empty  */
+    public boolean isEmpty() { return (m_first == null); }
+
+    /*  Get the current size of the list  */

```

Comment:
Does it matters?
Yes, usually in the method I use // and for longer descriptions /* */ as method description before.
But if you prefer all to be /* */, I can change it. np.